spec/components/app_session_needs_review_warning_component_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+describe AppSessionNeedsReviewWarningComponent do
+  subject(:rendered) { render_inline(component) }
+
+  let(:component) { described_class.new(session:) }
+  let(:session) { create(:session) }
+
+  describe "#render?" do
+    context "when session has no patients without NHS number" do
+      it "does not render the component" do
+        expect(component.render?).to be(false)
+      end
+
+      it "renders nothing" do
+        expect(rendered.to_html).to be_empty
+      end
+    end
+
+    context "when session has patients without NHS number" do
+      before { create(:patient, nhs_number: nil, session:) }
+
+      it "renders the component" do
+        expect(component.render?).to be(true)
+      end
+
+      it "renders the warning callout" do
```

Comment:
I wouldn't expect this test to be in the `describe "#render?"` block as it's not testing that method.